airflow/models/dagrun.py

```diff
@@ -621,10 +622,20 @@ def find_duplicate(cls, dag_id: str, run_id: str, *, session: Session = NEW_SESS
         return session.scalars(select(cls).where(cls.dag_id == dag_id, cls.run_id == run_id)).one_or_none()
 
     @staticmethod
-    def generate_run_id(run_type: DagRunType, logical_date: datetime) -> str:
-        """Generate Run ID based on Run Type and logical Date."""
+    def generate_run_id(
+        *, run_type: DagRunType, logical_date: datetime | None = None, run_after: datetime
+    ) -> str:
+        """
+        Generate Run ID based on Run Type, run_after and logical Date.
+
+        :param run_type: type of DagRun
+        :param logical_date: the logical date
+        :param run_after: the date before which dag run won't start.
+        """
         # _Ensure_ run_type is a DagRunType, not just a string from user code
-        return DagRunType(run_type).generate_run_id(logical_date)
+        if logical_date:
+            return DagRunType(run_type).generate_run_id(run_after=logical_date.isoformat())
```

Comment:
Is it correct to pass logical_date as run_after? Wouldn't it be confusing? when `logical_dates` exists, we pass it as `run_after` while when `run_after` exists, we pass `run_after + random string`

Reply:
As per discussion with @uranusjr When we have logical_date, we should generate run_id based on logical_date.

But when there is no run_after we should pass `run_after + random string`. 

In generate_run_id of DagRunType, I have just added one date field and I have called it `run_after` for now.

Reply:
I am not sure we should call it `run_after`. looks more like `suffix` to me. as long as @uranusjr 's good with it and users won't access it, I think we're good to keep it as it is

Reply:
I can rename this argument in another PR with a better name to unblock other PRs. WDYT?

Reply:
I get what @Lee-W is trying to say. The argument on DagRunType.generate_run_id is a str so yeah it’s probably better for it to be named `suffix`. On the timetable it’s a datetime (and does not include the random string) so `run_after` is more appropriate.

Reply:
Alternatively (to `run_after`) we can probably use an even vaguer word such as simply `date`.

Reply:
Changed it